app/renderer/components/common/contextMenu/contextMenuItem.js

```diff
@@ -88,28 +104,21 @@ class ContextMenuItem extends ImmutableComponent {
   }
 
   onMouseEnter (e) {
-    let openedSubmenuDetails = this.props.contextMenuDetail.get('openedSubmenuDetails')
-    openedSubmenuDetails = openedSubmenuDetails
-      ? openedSubmenuDetails.splice(this.props.submenuIndex, this.props.contextMenuDetail.get('openedSubmenuDetails').size)
-      : new Immutable.List()
-
-    if (this.hasSubmenu) {
-      let node = e.target
-      while (node && !elementHasDataset(node, 'contextMenuItem')) {
-        node = node.parentNode
-      }
-      let parentNode = node.parentNode
-      while (parentNode && !elementHasDataset(parentNode, 'contextMenu')) {
-        parentNode = parentNode.parentNode
-      }
-      const parentBoundingRect = parentNode.getBoundingClientRect()
-      const boundingRect = node.getBoundingClientRect()
+    if (this.props.contextMenuItem.has('folderKey')) {
+      const yAxis = this.getYAxis(e)
+      windowActions.onShowBookmarkFolderMenu(this.props.contextMenuItem.get('folderKey'), yAxis, null, this.props.submenuIndex)
+    } else if (this.hasSubmenu) {
+      let openedSubmenuDetails = this.props.contextMenuDetail.get('openedSubmenuDetails')
+      openedSubmenuDetails = openedSubmenuDetails
+        ? openedSubmenuDetails.splice(this.props.submenuIndex, this.props.contextMenuDetail.get('openedSubmenuDetails').size)
```

Comment:
`this.props.contextMenuDetail.get('openedSubmenuDetails')` is just fetched above; if it's truthy, we should be able to change the splice to re-use that variable.  So like:
`openedSubmenuDetails.splice(this.props.submenuIndex, openedSubmenuDetails.size)`

Reply:
done